src/Microsoft.Health.Fhir.Api/Controllers/AadSmartOnFhirProxyController.cs

```diff
@@ -304,6 +303,23 @@ public AadSmartOnFhirProxyController(IOptions<SecurityConfiguration> securityCon
             var client = _httpClientFactory.CreateClient();
 #pragma warning restore CA2000
 
+            if (Request.Headers.TryGetValue("Authorization", out var authHeaderValues)
+                && string.IsNullOrEmpty(clientId))
```

Comment:
What should we do if the client passes an auth header and a client_id/client_secret form value?

Reply:
As currently written, if the clientId is specified in the form value, then we will prioritize that value and ignore the auth header.   I suppose that is a valid approach, but am open to other suggestions.